### ISSUE TITLE
high contrast theming

### ISSUE THREAD

**Issue by akosyakov** [eclipse-theia/theia#6743]
Review how VS Code applies high contrast colors https://code.visualstudio.com/api/references/theme-color#contrast-colors and apply them similarly in Theia.

We leave it out from the initial theming.

**Comment by kittaakos** [comment]
Some issues are also listed here: https://github.com/eclipse-theia/theia/issues/7353
 - selection in the `Explorer`,
 - border of the buttons in the dialogs,
 - tbd

**Comment by kittaakos** [comment]
The context menu, in the browser example, is also broken:

![Screen Shot 2020-03-18 at 09 08 05](https://user-images.githubusercontent.com/1405703/76939349-81f95280-68f8-11ea-94f5-f25f7dde73eb.jpg)

**Comment by kittaakos** [comment]
[nitpick]: I think the border of the left side-bar is not aligned with the main menu bar:
![Screen Shot 2020-03-18 at 09 13 25](https://user-images.githubusercontent.com/1405703/76939552-def50880-68f8-11ea-9b5d-d23925f8ee1b.jpg)

**Comment by kaiyue0329** [comment]
In the Problems View, there is no border around the selected node.

In VS code,
![image](https://user-images.githubusercontent.com/23107734/77187786-62565b80-6aab-11ea-8c07-b137cdd2784d.png)

In Theia,
![image](https://user-images.githubusercontent.com/23107734/77188083-d264e180-6aab-11ea-8b0e-fc319683d29c.png)
(Arrow pointing to the selected node)

**Comment by kittaakos** [comment]
Editor and Explorer decorators always white with the HC theme. I would expect red and yellowish colors.
![Screen Shot 2020-04-02 at 16 13 43](https://user-images.githubusercontent.com/1405703/78259439-01903f80-74fd-11ea-9afc-3148a4f3b3f8.jpg)

**Comment by kittaakos** [comment]
Also: https://github.com/eclipse-theia/theia/issues/7489

**Comment by kittaakos** [comment]
Support for hover. VS Code:
![screencast 2020-04-02 17-21-32](https://user-images.githubusercontent.com/1405703/78269368-c6483d80-7509-11ea-9ba1-cd9dfe5eee21.gif)

TODO: Revisit whether we want to support it for dialog buttons or not. VS Code does not have any dialogs but they do not support it in the Explorer:

![screencast 2020-04-02 17-47-00](https://user-images.githubusercontent.com/1405703/78269664-30f97900-750a-11ea-8dae-31815075e653.gif)

**Comment by kittaakos** [comment]
Difference between the editor error markers:
VS Code
![Screen Shot 2020-04-02 at 18 15 48](https://user-images.githubusercontent.com/1405703/78272707-2b9e2d80-750e-11ea-9c3a-20cc6aa40b6f.jpg)


Theia:
![Screen Shot 2020-04-02 at 18 15 57](https://user-images.githubusercontent.com/1405703/78272689-25a84c80-750e-11ea-97e6-96f67d250d71.jpg)



Maybe it is a TS issue.
